astropy/wcs/wcsapi/tests/test_fitswcs.py

```diff
@@ -445,23 +447,278 @@ def test_time_cube():
-    with warnings.catch_warnings(record=True) as warning_entries:
-        warnings.resetwarnings()
-        coord, time = wcs.pixel_to_world(29, 39, 44)
-
-    # Check that there's at least one warning of the right category/message
-    assert len(warning_entries) > 0
-    found_warning = False
-    for w in warning_entries:
-        msg = 'In future, times will be represented by the Time class'
-        if w.category is FutureWarning and str(w.message).startswith(msg):
-            found_warning = True
-    assert found_warning
+    coord, time = wcs.pixel_to_world(29, 39, 44)
+    assert isinstance(coord, SkyCoord)
+    assert isinstance(coord.frame, ICRS)
+    assert_allclose(coord.ra.deg, 1.7323356692202325)
+    assert_allclose(coord.dec.deg, 14.783516054817797)
+    assert isinstance(time, Time)
+    assert_allclose(time.mjd, 54746.03429755324)
 
+    coord, time = wcs.array_index_to_world(44, 39, 29)
     assert isinstance(coord, SkyCoord)
+    assert isinstance(coord.frame, ICRS)
+    assert_allclose(coord.ra.deg, 1.7323356692202325)
+    assert_allclose(coord.dec.deg, 14.783516054817797)
+    assert isinstance(time, Time)
+    assert_allclose(time.mjd, 54746.03429755324)
+
+    x, y, z = wcs.world_to_pixel(coord, time)
+    assert_allclose(x, 29.)
+    assert_allclose(y, 39.)
+    assert_allclose(z, 44.)
+
+    # Order of world coordinates shouldn't matter
+    x, y, z = wcs.world_to_pixel(time, coord)
+    assert_allclose(x, 29.)
+    assert_allclose(y, 39.)
+    assert_allclose(z, 44.)
+
+    i, j, k = wcs.world_to_array_index(coord, time)
+    assert_equal(i, 44)
+    assert_equal(j, 39)
+    assert_equal(k, 29)
+
+    # Order of world coordinates shouldn't matter
+    i, j, k = wcs.world_to_array_index(time, coord)
+    assert_equal(i, 44)
+    assert_equal(j, 39)
+    assert_equal(k, 29)
+
+###############################################################################
+# The following tests are to make sure that Time objects are constructed
+# correctly for a variety of combinations of WCS keywords
+###############################################################################
+
+
+HEADER_TIME_1D = """
+SIMPLE  = T
+BITPIX  = -32
+NAXIS   = 1
+NAXIS1  = 2048
+TIMESYS = 'UTC'
+TREFPOS = 'TOPOCENT'
+MJDREF  = 50002.6
+CTYPE1  = 'UTC'
+CRVAL1  = 5
+CUNIT1  = 's'
+CRPIX1  = 1.0
+CDELT1  = 2
+OBSGEO-L= -20
+OBSGEO-B= -70
+OBSGEO-H= 2530
+"""
+
+
+def assert_time_at(header, position, jd1, jd2, scale, format):
+    wcs = WCS(header)
+    time = wcs.pixel_to_world(position)
+    assert_allclose(time.jd1, jd1, rtol=1e-10)
+    assert_allclose(time.jd2, jd2, rtol=1e-10)
+    assert time.format == format
+    assert time.scale == scale
+
+
+def test_time_1d_values():
+
+    # Check that Time objects are instantiated with the correct values,
+    # scales, and formats.
+
+    header = Header.fromstring(HEADER_TIME_1D, sep='\n')
+
+    # Main supported time scales
+    for scale in ('tai', 'tcb', 'tcg', 'tdb', 'tt', 'ut1', 'utc', 'local'):
```

Comment:
Not really important, but this test suite seems to cry out for a class with a `setup` method that initializes the header and `pytest.mark.parametrize` to go through the scales.

Reply:
```python

@pytest.fixture(params=('tai', 'tcb', 'tcg', 'tdb', 'tt', 'ut1', 'utc', 'local'))
def time_header_with_scale(request):
    header = Header.fromstring(HEADER_TIME_1D, sep='\n')
    header['CTYPE1'] = request.param.upper()
    return header
```

Reply:
Done!